### ISSUE TITLE
Tags an Antworten neu vergeben

### ISSUE THREAD

**Issue by campingrider** [ckruse/cforum#521]
Ich habe neulich das "Tags auch an Antworten neu vergeben" Feature genutzt und dabei ist mir aufgefallen, dass es nicht besonders sinnig ist, wenn die Tags tatsächlich an alle Antworten vergeben werden. Viel sinnvoller wäre es mMn, wenn die Tags der Antworten nur dann mitgeändert würden, wenn sie nicht vom Autor der Antwort schon absichtlich abgeändert worden sind, d.h. automatische Änderung nur bei den Antworten, die dasselbe Set von Tags haben. Ansonsten macht man jeden Versuch eines Autors, Threaddrift zu kennzeichnen (was ja sinnvoll ist) wieder zunichte, nur weil man der Ausgangsfrage und den thematisch passenden Antworten ein neues Tag verpassen wollte...

**Comment by campingrider** [comment]
Könnte natürlich auch als weitere Option angeboten werden, ich denke aber, dass die ursprüngliche Option keinerlei Usecase hat.

**Comment by auge8472** [comment]
Natürlich ist es möglich, eine absichtliche Änderung der Tags zu überschreiben. Diese Gefahr besteht. Sie ist aber genauso eine Chance auf eine besseres Tagging eines Threads, derer man sich ohne diese Möglichkeit berauben würde.

Ich halte es allerdings für zumutbar, dass jener, der die Tags ergänzt oder neu vergibt, die Konsequenzen vor dem Abschicken der neuen Tags zu bedenken. Das darf ja nicht jeder x-beliebige Benutzer. Die, "die sich das verdient haben", sollten mMn durchaus so umsichtig sein, das zu berücksichtigen.

**Comment by campingrider** [comment]
Nein, wenn eine von 5 Antworten die Tags entfernt und auf Menschelei geändert hat ist dieser Subthread eben nicht besser getaggt, wenn ein Admin die Tags des Threads verändert hat. Ich schlage ja nicht vor, das Feature zu entfernen, sondern nur das Feature smarter zu gestalten. Es geht auch nicht um Umsichtigkeit, sondern um Machbarkeit. Ich kann als Admin bei einer nötigen Tagänderung bei einem elends langen Thread nicht jeden Subthread einzeln auf Tagänderungen prüfen und umtaggen. Ergo ist es also so, dass man das Feature im Moment gar nicht sinnvoll einzusetzen ist, weil ich riskiere sinnvolle Tagänderungen damit zu überfahren - und dann lass ichs doch lieber ganz.

**Comment by auge8472** [comment]
Ich verstehe, worauf du hinaus willst. Ich sehe nur nicht das Problem.

Natürlich sollte man nicht _einen ganzen Thread_ retaggen, wenn man schon sieht, dass es dort reihenweise Veränderungen, also bewusst gesetzte neue Tags, gibt. Dann macht man sich die Arbeit, Threadäste zu retaggen, nur das betreffende Posting zu retaggen oder man lässt es ganz.

Es gibt aber durchaus oft die Fälle von Threads, die vom Ersteller völlig am Thema vorbei getaggt wurden. Da kann man als berechtigter Benutzer – die Funktion ist ja nicht allein den Moderatoren und Admins vorbehalten – die Tags neu setzen und auch an die in diesen Fällen normalerweise erst wenigen Antworten mit einbeziehen. Das habe zumindest ich schon einige Male getan. Soweit ich mich erinnere, wurden eventuell in Antworten bereits abweichend von denen des Eröffnungspostings vergebene Tags nur ergänzt. Ob das noch so funktioniert, wenn eines der Antwortpostings schon von dessen Autor die maximale Anzahl von drei Tags verpasst bekommen hat, weiß ich allerdings nicht.

**Comment by campingrider** [comment]
Naja, du nennst das Problem im Grunde schon:

> Natürlich sollte man nicht *einen ganzen Thread* retaggen, wenn man schon sieht, dass es dort reihenweise Veränderungen, also bewusst gesetzte neue Tags, gibt. Dann macht man sich die Arbeit [das manuell zu machen] oder man lässt es ganz.

Also, das ist doch genau das Problem. Es gibt ein Feature zur Automatisierung, es ist aber meist nicht sinnvoll zu gebrauchen. Lass mich ausholen:

Es gibt das Feature "Tags auch an Antworten neu vergeben". Effektiv ist es aber in der Form, in der es momentan existiert, sinnlos. Denn entweder man kann es nicht benutzen, weil zu viele Antworten eingegangen sind und muss dann

> sich die Arbeit, Threadäste zu retaggen [, machen]

oder es gibt

> erst wenige[n] Antworten

und man kann das Feature nutzen, aber da könnte man das, um den Spieß mal umzudrehen, genauso von Hand machen.

In der momentanen Situation nimmt das Feature einem also nur dann Arbeit ab (indem automatisch mehrere Posts retaggt werden), wenn die Gesamtmenge an zu editierenden Posts gering ist - d.h. in einer Situation, in der das Arbeit abnehmen eigentlich noch gar nicht so nötig wäre.

Mein Wunsch ist, dass das Feature einem die Arbeit auch dann abnimmt, wenn die Gesamtmenge an zu editierenden Posts größer ist, also an einer Stelle, die geradezu danach schreit, die Arbeit zu automatisieren.

Und dann kommen noch zwei Dinge dazu:

1. Es gibt keine (für mich denkbare) Situation, in der das jetzige Feature dem Feature so wie ich es mir wünsche überlegen wäre.
2. Der Aufwand zur Änderung des Features ist denkbar gering (es geht im Zweifelsfall nur darum, bei jedem zu ändernden Posting eine Abfrage mit einzubauen; ich weiß nicht wie die Implementierung im Moment aussieht, wage aber zu behaupten, dass dazu lediglich eine Zeile SQL angepasst werden muss)

Und last but not least: Der Schaden, der bei einem fälschlichen Gebrauch des Features in seiner momentanen Form entsteht ist - innerhalb dessen, was beim Taggen überhaupt "kaputt gehen" kann - maximal, weil potenziell etliche Postings ihre sinnvollen Zuordnungen verlieren. Bei einer Änderung des Features wird dieser potenzielle Schaden geringer. Und, wie gesagt, die Änderung kostet nicht viel und ist der momentanen Funktion überlegen. 

Es ist ja nicht so, dass das Feature in seiner momentanen Form *kaputt* wäre. Es ist nur so, dass es in seiner aktuellen Form **weit** hinter seinem Potenzial zurückbleibt, was die Arbeitserleichterung angeht.

**Comment by MatthiasApsel** [comment]
Auch das Hinzufügen oder Löschen einzelner Tags birgt dieselben Probleme wie das Ändern der kompletten Taglist.

Beispiel: Das OP ist eine Frage zu SQL fälschlicherweise mit CSS und HTML getagt. Inzwischen gibt es aber einen Menschelei-Ast und einen Ast, der auf die Umsetzung des Formulars eingeht (HTML), in welchem es auch um die Gestaltung geht (HTML + CSS).

Für den Menschelei-Ast würdest du nun das Tag SQL hinzufügen, der da aber nichts zu suchen hat, dem zweiten Ast würdest du das HTML wegnehmen, was falsch wäre, ebenso für den Gestaltungsast.

Auch der Vorschlag, Tags nur dann zu ändern, wenn das ganze Tagset übereinstimmt, ist ggf. nicht zielführend, denn der Ast mit der Gestaltung ist mit HTML + CSS korrekt ausgezeichnet. Das würde jetzt mit SQL getagt.

Ich sehe keine Möglichkeit, den vorhandenen Automatismus so zu erweitern, dass er mehr nützt als schadet.

**Comment by campingrider** [comment]
> Auch der Vorschlag, Tags nur dann zu ändern, wenn das ganze Tagset übereinstimmt, ist ggf. nicht zielführend, denn der Ast mit der Gestaltung ist mit HTML + CSS korrekt ausgezeichnet. Das würde jetzt mit SQL getagt.

Aber deutlich zielführender als momentan. Es sollte natürlich nach Möglichkeit so gemacht werden, dass Kind-Postings eines Postings mit nicht-übereinstimmendem Tagset auch nicht geändert werden.

Zwar gibt es dann immer noch Szenarios, in denen Äste zu Unrecht retaggt werden, aber deutlich weniger. Und in diesen Fällen betrifft das dann auch nur die erste Ebene, weil ab der zweiten Ebene ja wieder unter Nutzung des selben Features zurückgeändert werden kann.

> Ich sehe keine Möglichkeit, den vorhandenen Automatismus so zu erweitern, dass er mehr nützt als schadet.

Doch, mit der Änderung nützt er schon mehr als er schadet, bzw. er nützt dann zumindest mehr als er jetzt nützt.

Alle Problem-Szenarios auszuschließen ist durch Erweiterung des vorhandenen Automatismus nicht möglich, da gebe ich dir recht. Dazu müsste man eine Möglichkeit haben, beispielsweise mit einer verschachtelten Liste von Checkboxen, die Äste, die mitgeändert werden sollen, manuell auszuwählen. Das ist dann wirklich idiotensicher, aber eben aufwändiger als die Erweiterung des vorhandenen Automatismus um eine Abfrage ;)
